thunder/executors/torchex.py

```diff
@@ -145,16 +145,10 @@ def _device_put_transform(a: TensorProxy, device: devices.Device) -> TensorProxy
 #
 
 
-class no_autocast(ContextDecorator):
-    def __enter__(self, *args, **kwargs):
-        self.was_cuda_enabled = torch.is_autocast_enabled()
-        self.was_cpu_enabled = torch.is_autocast_cpu_enabled()
-        torch.set_autocast_enabled(False)
-        torch.set_autocast_cpu_enabled(False)
-
-    def __exit__(self, exc_type, exc_value, exc_traceback):
-        torch.set_autocast_enabled(self.was_cuda_enabled)
-        torch.set_autocast_cpu_enabled(self.was_cpu_enabled)
+def no_autocast(fn):
+    fn = torch.autocast(device_type="cpu", enabled=False, cache_enabled=False)(fn)
+    fn = torch.autocast(device_type="cuda", enabled=False, cache_enabled=False)(fn)
+    return fn
```

Comment:
Would the following work equivalently but with the effect that there would be zero change for the notebooks?
```py
def no_autocast():
    def decorator(fn):
        fn = torch.autocast(device_type="cpu", enabled=False, cache_enabled=False)(fn)
        fn = torch.autocast(device_type="cuda", enabled=False, cache_enabled=False)(fn)
        return fn
    return decorator
```

Reply:
I can try this but from my experience, `torch.compile` has poor support for generic decorators

Reply:
It does work. Do you want me to revert this change?

Reply:
No, I was checking my understanding.